File: reference source/xml2scanner/XML2Scanner/_package.vb
Public Class _package
    ' Package ID

    Private m_pkgid As String

    Public Property CPkgID() As String

        Get

            Return m_pkgid

        End Get

        Set(ByVal value As String)

            m_pkgid = value

        End Set

    End Property

    ' Package Name

    Private m_pkgname As String

    Public Property CPackageName() As String

        Get

            Return m_pkgname

        End Get

        Set(ByVal value As String)

            m_pkgname = value

        End Set

    End Property

    ' Package Version

    Private m_version As String

    Public Property CVersion() As String

        Get

            Return m_version

        End Get

        Set(ByVal value As String)

            m_version = value

        End Set

    End Property

    ' Package Type

    Private m_pkgtype As String

    Public Property CPackageType() As String

        Get

            Return m_pkgtype

        End Get

        Set(ByVal value As String)

            m_pkgtype = value

        End Set

    End Property

    ' Package Reboot

    Private m_reboot As String

    Public Property CReboot() As String

        Get

            Return m_reboot

        End Get

        Set(ByVal value As String)

            m_reboot = value

        End Set

    End Property

    ' Package Brand

    Private m_brand As String

    Public Property CBrand() As String

        Get

            Return m_brand

        End Get

        Set(ByVal value As String)

            m_brand = value

        End Set

    End Property

    ' Package Severity

    Private m_severity As String

    Public Property CSeverity() As String

        Get

            Return m_severity

        End Get

        Set(ByVal value As String)

            m_severity = value

        End Set

    End Property

    ' Package Title

    Private m_title As String

    Public Property CTitle() As String

        Get

            Return m_title

        End Get

        Set(ByVal value As String)

            m_title = value

        End Set

    End Property

    ' Release Date

    Private m_released As Date

    Public Property CReleased() As Date

        Get

            Return m_released

        End Get

        Set(ByVal value As Date)

            m_released = value

        End Set

    End Property

    ' URL to XML2
    Private m_urlxml2 As String

    Public Property CURLxml2() As String

        Get

            Return m_urlxml2

        End Get

        Set(ByVal value As String)

            m_urlxml2 = value

        End Set

    End Property

    ' Category
    Private m_category As String

    Public Property CCategory() As String

        Get

            Return m_category

        End Get

        Set(ByVal value As String)

            m_category = value

        End Set

    End Property

    ' URL to EXE
    Private m_urlexe As String

    Public Property CURLexe() As String

        Get

            Return m_urlexe

        End Get

        Set(ByVal value As String)

            m_urlexe = value

        End Set

    End Property

    ' CRC of EXE
    Private m_crcexe As String

    Public Property CCRCexe() As String

        Get
            Return m_crcexe
        End Get

        Set(ByVal value As String)
            m_crcexe = value
        End Set

    End Property

    'URL to Readme TXT
    Private m_urltxt As String

    Public Property CURLtxt() As String

        Get
            Return m_urltxt
        End Get

        Set(ByVal value As String)
            m_urltxt = value
        End Set

    End Property


    'XML2 crc
    Private m_xml2crc As String

    Public Property Cxml2crc() As String

        Get

            Return m_xml2crc

        End Get

        Set(ByVal value As String)

            m_xml2crc = value

        End Set

    End Property

    'XML2 CRC actual
    Private m_xml2crcactual As String
    Public Property Cxml2crcactual() As String

        Get

            Return m_xml2crcactual

        End Get

        Set(ByVal value As String)

            m_xml2crcactual = value

        End Set

    End Property

    'XML2 crc check
    Private m_xml2valid As String

    Public Property Cxml2valid() As String

        Get

            Return m_xml2valid

        End Get

        Set(ByVal value As String)

            m_xml2valid = value

        End Set

    End Property

    'EXE crc check
    Private m_exevalid As String

    Public Property Cexevalid() As String

        Get

            Return m_exevalid

        End Get

        Set(ByVal value As String)

            m_exevalid = value

        End Set

    End Property

    'Language tags
    Private m_langs As String

    Public Property Clangs() As String

        Get

            Return m_langs

        End Get

        Set(ByVal value As String)

            m_langs = value

        End Set

    End Property

    'Setup command
    Private m_setup As String
    Public Property Csetup() As String

        Get

            Return m_setup

        End Get

        Set(ByVal value As String)

            m_setup = value

        End Set

    End Property
End Class

File: investigator/investigator/_package.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace investigator
{
    class _package
    {
        // Package ID

        private string m_pkgid;

        public string CPkgID
        {

            get
            {

                return m_pkgid;

            }

            set
            {

                m_pkgid = value;

            }

        }

        // Package Name

        private string m_pkgname;

        public string CPackageName
        {

            get
            {

                return m_pkgname;

            }

            set
            {

                m_pkgname = value;

            }

        }

        // Package Version

        private string m_version;

        public string CVersion
        {

            get
            {

                return m_version;

            }

            set
            {

                m_version = value;

            }

        }

        // Package Type

        private string m_pkgtype;

        public string CPackageType
        {

            get
            {

                return m_pkgtype;

            }

            set
            {

                m_pkgtype = value;

            }

        }

        // Package Reboot

        private string m_reboot;

        public string CReboot
        {

            get
            {

                return m_reboot;

            }

            set
            {

                m_reboot = value;

            }

        }

        // Package Brand

        private string m_brand;

        public string CBrand
        {

            get
            {

                return m_brand;

            }

            set
            {

                m_brand = value;

            }

        }

        // Package Severity

        private string m_severity;

        public string CSeverity
        {

            get
            {

                return m_severity;

            }

            set
            {

                m_severity = value;

            }

        }

        // Package Title

        private string m_title;

        public string CTitle
        {

            get
            {

                return m_title;

            }

            set
            {

                m_title = value;

            }

        }

        // Release Date

        private DateTime m_released;

        public DateTime CReleased
        {

            get
            {

                return m_released;

            }

            set
            {

                m_released = value;

            }

        }

        // URL to XML2
        private string m_urlxml2;

        public string CURLxml2
        {

            get
            {

                return m_urlxml2;

            }

            set
            {

                m_urlxml2 = value;

            }

        }

        // Category
        private string m_category;

        public string CCategory
        {

            get
            {

                return m_category;

            }

            set
            {

                m_category = value;

            }

        }

        // URL to EXE
        private string m_urlexe;

        public string CURLexe
        {

            get
            {

                return m_urlexe;

            }

            set
            {

                m_urlexe = value;

            }

        }

        // CRC of EXE
        private string m_crcexe;

        public string CCRCexe
        {

            get
            {
                return m_crcexe;
            }

            set
            {
                m_crcexe = value;
            }

        }

        // URL to Readme TXT
        private string m_urltxt;

        public string CURLtxt
        {

            get
            {
                return m_urltxt;
            }

            set
            {
                m_urltxt = value;
            }

        }


        // XML2 crc
        private string m_xml2crc;

        public string Cxml2crc
        {

            get
            {

                return m_xml2crc;

            }

            set
            {

                m_xml2crc = value;

            }

        }

        // XML2 CRC actual
        private string m_xml2crcactual;
        public string Cxml2crcactual
        {

            get
            {

                return m_xml2crcactual;

            }

            set
            {

                m_xml2crcactual = value;

            }

        }

        // XML2 crc check
        private string m_xml2valid;

        public string Cxml2valid
        {

            get
            {

                return m_xml2valid;

            }

            set
            {

                m_xml2valid = value;

            }

        }

        // EXE crc check
        private string m_exevalid;

        public string Cexevalid
        {

            get
            {

                return m_exevalid;

            }

            set
            {

                m_exevalid = value;

            }

        }

        // Language tags
        private string m_langs;

        public string Clangs
        {

            get
            {

                return m_langs;

            }

            set
            {

                m_langs = value;

            }

        }

        // Setup command
        private string m_setup;
        public string Csetup
        {

            get
            {

                return m_setup;

            }

            set
            {

                m_setup = value;

            }

        }

    }
}
